Assets/Mirror/Runtime/NetworkManager.cs

```diff
@@ -315,11 +311,40 @@ public bool StartServer()
             }
             else
             {
-                NetworkServer.SpawnObjects();
+                // spawn objects, but only if not starting server as host mode.
+                // spawning objects calls OnStartServer
+                //   -> OnStartServer might spawn an object and set [SyncVar(hook="OnColorChanged")] object.color = green;
+                //      -> this calls SyncVar.set (generated by Weaver), which has
+                //         a custom case for host mode (because host mode doesn't
+                //         get OnDeserialize calls, where SyncVar hooks are usually
+                //         called):
+                //
+                //           if (!SyncVarEqual(value, ref color))
+                //           {
+                //               if (NetworkServer.localClientActive && !getSyncVarHookGuard(1uL))
+                //               {
+                //                   setSyncVarHookGuard(1uL, value: true);
+                //                   OnColorChangedHook(value);
+                //                   setSyncVarHookGuard(1uL, value: false);
+                //               }
+                //               SetSyncVar(value, ref color, 1uL);
+                //           }
+                //
+                //    => so we need to call OnStartServer AFTER the host
+                //       connected.
+                if (automaticallySpawnObjects)
+                    NetworkServer.SpawnObjects();
             }
             return true;
         }
 
+        /// <summary>
+        /// This starts a new server.
+        /// <para>This uses the networkPort property as the listen port.</para>
+        /// </summary>
+        /// <returns></returns>
+        public bool StartServer() => StartServerInternal(true);
```

Comment:
```suggestion
        public bool StartServer() {
            StartServerInternal();
            NetworkServer.SpawnObjects();
        }
```

How about StartServerInternal does not receive any parameter,  and just acts as you always pass null,  then you simply call SpawnObjects afterwards